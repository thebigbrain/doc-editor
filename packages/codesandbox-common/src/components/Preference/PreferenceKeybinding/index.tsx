import React from 'react';
import KeybindingInput from './KeybindingInput';

export type Props = {
  setValue: (value: Array<string[]>) => void;
  value: Array<string[]>;
};

export default class PreferenceKeybinding extends React.PureComponent<Props> {
  setValue = index => value => {
    const result = [...this.props.value];
    result[index] = value;

    this.props.setValue(result);
  };

  render() {
    const { value } = this.props;

    return (
      <div>
        <KeybindingInput
          {...this.props}
          placeholder="First"
          value={value[0]}
          setValue={this.setValue(0)}
        />
        {' - '}
        <KeybindingInput
          {...this.props}
          placeholder="Second"
          value={value.length === 2 && value[1]}
          setValue={this.setValue(1)}
          disabled={!value[0] || value[0].length === 0}
        />
      </div>
    );
  }
}
