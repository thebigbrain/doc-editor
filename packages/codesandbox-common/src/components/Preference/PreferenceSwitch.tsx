import React from 'react';
import Switch from '../Switch';

export type Props = {
  value: boolean;
  setValue: (val: boolean) => void;
};

export default class PreferenceSwitch extends React.Component<Props> {
  handleClick = () => {
    this.props.setValue(!this.props.value);
  };

  render() {
    const { value } = this.props;
    return (
      <Switch
        onClick={this.handleClick}
        small
        style={{ width: '3rem' }}
        offMode
        secondary
        right={value}
      />
    );
  }
}
