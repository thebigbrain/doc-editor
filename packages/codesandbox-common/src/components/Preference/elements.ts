import styled from 'styled-components';
import Input from '../Input';

export const Container = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const StyledInput = styled(Input)`
  text-align: center;
`;
