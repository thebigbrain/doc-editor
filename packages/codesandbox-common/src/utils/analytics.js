"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
exports.__esModule = true;
var version_1 = require("../version");
var debug_1 = require("../utils/debug");
var hash_1 = require("./hash");
var debug = debug_1["default"]('cs:analytics');
var global = (typeof window !== 'undefined' ? window : {});
var WHITELISTED_VSCODE_EVENTS = [
    'codesandbox.preview.toggle',
    'workbench.action.splitEditor',
    'workbench.action.toggleSidebarVisibility',
    'codesandbox.sandbox.new',
    'workbench.action.files.saveAs',
    'editor.action.addCommentLine',
    'codesandbox.sandbox.exportZip',
    'codesandbox.preferences',
    'codesandbox.sandbox.fork',
    'codesandbox.help.documentation',
    'codesandbox.help.github',
    'view.preview.flip',
    'codesandbox.search',
    'workbench.action.splitEditorLeft',
    'codesandbox.dashboard',
    'workbench.action.toggleCenteredLayout',
    'workbench.action.toggleMenuBar',
    'codesandbox.explore',
    'editor.action.toggleTabFocusMode',
    'workbench.action.splitEditorUp',
    'workbench.action.toggleSidebarPosition',
    'workbench.action.toggleActivityBarVisibility',
    'workbench.action.toggleStatusbarVisibility',
    'codesandbox.dependencies.add',
    'codesandbox.help.open-issue',
    'codesandbox.action.search',
    'workbench.action.editorLayoutThreeColumns',
    'breadcrumbs.toggleToOn',
    'workbench.action.openSettings2',
    'workbench.action.globalSettings',
    'workbench.action.editorLayoutTwoRows',
    'workbench.action.editorLayoutTwoByTwoGrid',
    'editor.action.showContextMenu',
    'toggleVim',
    'codesandbox.help.spectrum',
    'codesandbox.help.feedback',
    'workbench.action.webview.openDeveloperTools',
    'workbench.action.editorLayoutThreeRows',
    'codesandbox.help.twitter',
    'workbench.action.editorLayoutTwo',
    'codesandbox.preview.external',
    'notifications.showList',
    'workbench.action.editor.changeEncoding',
    'editor.action.indentationToTabs',
    'workbench.action.maximizeEditor',
    'editor.action.indentationToSpaces',
    'revealFilesInOS',
    'keybindings.editor.searchKeyBindings',
    'notifications.hideList',
    'workbench.action.terminal.focus',
    'workbench.action.console.focus',
    'workbench.action.openRecent',
    'code-runner.run',
];
exports.DNT = typeof window !== 'undefined' &&
    Boolean(global.doNotTrack === '1' ||
        global.navigator.doNotTrack === '1' ||
        global.navigator.msDoNotTrack === '1');
var sentryInitialized = false;
function getSentry() {
    return Promise.resolve().then(function () { return require(/* webpackChunkName: 'sentry' */ '@sentry/browser'); });
}
function initializeSentry(dsn) {
    return __awaiter(this, void 0, void 0, function () {
        var Sentry;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    if (!!exports.DNT) return [3 /*break*/, 2];
                    sentryInitialized = true;
                    return [4 /*yield*/, getSentry()];
                case 1:
                    Sentry = _a.sent();
                    return [2 /*return*/, Sentry.init({
                            dsn: dsn,
                            release: version_1["default"],
                            ignoreErrors: [
                                'Custom Object',
                                'TypeScript Server Error',
                                /^Canceled$/,
                            ],
                            /**
                             * Don't send messages from the sandbox, so don't send from eg.
                             * new.codesandbox.io or new.csb.app
                             */
                            blacklistUrls: [
                                'codesandbox.editor.main.js',
                                /.*\.codesandbox\.io/,
                                /.*\.csb\.app/,
                            ],
                            beforeSend: function (event) {
                                if (event.stacktrace &&
                                    event.stacktrace.frames &&
                                    event.stacktrace.frames[0] &&
                                    event.stacktrace.frames[0].filename.endsWith('codesandbox.editor.main.js')) {
                                    // This is the spammy event that doesn't do anything: https://sentry.io/organizations/codesandbox/issues/1054971728/?project=155188&query=is%3Aunresolved
                                    // Don't do anything with it right now, I can't seem to reproduce it for some reason.
                                    // We need to add sourcemaps
                                    return undefined;
                                }
                                return event;
                            }
                        })];
                case 2: return [2 /*return*/, Promise.resolve()];
            }
        });
    });
}
exports.initializeSentry = initializeSentry;
function logError(err) {
    return __awaiter(this, void 0, void 0, function () {
        var Sentry;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    if (!sentryInitialized) return [3 /*break*/, 2];
                    return [4 /*yield*/, getSentry()];
                case 1:
                    Sentry = _a.sent();
                    Sentry.captureException(err);
                    _a.label = 2;
                case 2:
                    if (window.console && console.error)
                        console.error(err);
                    return [2 /*return*/];
            }
        });
    });
}
exports.logError = logError;
function identify(key, value) {
    return __awaiter(this, void 0, void 0, function () {
        var identity, Sentry, e_1;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    _a.trys.push([0, 3, , 4]);
                    if (!!exports.DNT) return [3 /*break*/, 2];
                    if (typeof global.amplitude !== 'undefined') {
                        identity = new global.amplitude.Identify();
                        identity.set(key, value);
                        global.amplitude.identify(identity);
                        debug('[Amplitude] Identifying', key, value);
                    }
                    if (!sentryInitialized) return [3 /*break*/, 2];
                    return [4 /*yield*/, getSentry()];
                case 1:
                    Sentry = _a.sent();
                    Sentry.configureScope(function (scope) {
                        scope.setExtra(key, value);
                    });
                    _a.label = 2;
                case 2: return [3 /*break*/, 4];
                case 3:
                    e_1 = _a.sent();
                    return [3 /*break*/, 4];
                case 4: return [2 /*return*/];
            }
        });
    });
}
exports.identify = identify;
if (process.env.NODE_ENV === 'production') {
    setTimeout(function () {
        identify('[Amplitude] Version', version_1["default"]);
    }, 5000);
}
function setUserId(userId) {
    return __awaiter(this, void 0, void 0, function () {
        var hashedId, Sentry, e_2;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    _a.trys.push([0, 3, , 4]);
                    if (!!exports.DNT) return [3 /*break*/, 2];
                    if (typeof global.amplitude !== 'undefined') {
                        hashedId = hash_1["default"](userId);
                        debug('[Amplitude] Setting User ID', hashedId);
                        identify('userId', hashedId);
                        global.amplitude.getInstance().setUserId(hashedId);
                    }
                    if (!sentryInitialized) return [3 /*break*/, 2];
                    return [4 /*yield*/, getSentry()];
                case 1:
                    Sentry = _a.sent();
                    Sentry.configureScope(function (scope) {
                        scope.setUser({ id: userId });
                    });
                    _a.label = 2;
                case 2: return [3 /*break*/, 4];
                case 3:
                    e_2 = _a.sent();
                    return [3 /*break*/, 4];
                case 4: return [2 /*return*/];
            }
        });
    });
}
exports.setUserId = setUserId;
function resetUserId() {
    return __awaiter(this, void 0, void 0, function () {
        var Sentry, e_3;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    _a.trys.push([0, 3, , 4]);
                    if (!!exports.DNT) return [3 /*break*/, 2];
                    if (typeof global.amplitude !== 'undefined') {
                        debug('[Amplitude] Resetting User ID');
                        identify('userId', null);
                        if (global.amplitude.getInstance().options.userId) {
                            global.amplitude.getInstance().setUserId(null);
                            global.amplitude.getInstance().regenerateDeviceId();
                        }
                    }
                    if (!sentryInitialized) return [3 /*break*/, 2];
                    return [4 /*yield*/, getSentry()];
                case 1:
                    Sentry = _a.sent();
                    Sentry.configureScope(function (scope) {
                        scope.setUser({ id: undefined });
                    });
                    _a.label = 2;
                case 2: return [3 /*break*/, 4];
                case 3:
                    e_3 = _a.sent();
                    return [3 /*break*/, 4];
                case 4: return [2 /*return*/];
            }
        });
    });
}
exports.resetUserId = resetUserId;
var isAllowedEvent = function (eventName, secondArg) {
    try {
        if (eventName === 'VSCode - workbenchActionExecuted') {
            return WHITELISTED_VSCODE_EVENTS.includes(secondArg.id);
        }
        return true;
    }
    catch (e) {
        return true;
    }
};
// After 30min no event we mark a session
var NEW_SESSION_TIME = 1000 * 60 * 30;
var getLastTimeEventSent = function () {
    var lastTime = localStorage.getItem('csb-last-event-sent');
    if (lastTime === null) {
        return 0;
    }
    return +lastTime;
};
var markLastTimeEventSent = function () {
    localStorage.setItem('csb-last-event-sent', Date.now().toString());
};
function track(eventName, secondArg) {
    if (secondArg === void 0) { secondArg = {}; }
    try {
        if (!exports.DNT && isAllowedEvent(eventName, secondArg)) {
            var data = __assign(__assign({}, secondArg), { version: version_1["default"], path: location.pathname + location.search });
            try {
                if (global.ga) {
                    global.ga('send', data);
                }
            }
            catch (e) {
                /* */
            }
            try {
                if (typeof global.amplitude !== 'undefined') {
                    var currentTime = Date.now();
                    if (currentTime - getLastTimeEventSent() > NEW_SESSION_TIME) {
                        // We send a separate New Session event if people have been inactive for a while
                        global.amplitude.logEvent('New Session');
                    }
                    markLastTimeEventSent();
                    debug('[Amplitude] Tracking', eventName, data);
                    global.amplitude.logEvent(eventName, data);
                }
            }
            catch (e) {
                /* */
            }
        }
    }
    catch (e) {
        /* empty */
    }
}
exports["default"] = track;
