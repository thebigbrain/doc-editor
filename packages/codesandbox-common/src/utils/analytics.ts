import VERSION from '../version';
import _debug from '../utils/debug';

import hash from './hash';

const debug = _debug('cs:analytics');

const global = (typeof window !== 'undefined' ? window : {}) as any;

const WHITELISTED_VSCODE_EVENTS = [
  'codesandbox.preview.toggle',
  'workbench.action.splitEditor',
  'workbench.action.toggleSidebarVisibility',
  'codesandbox.sandbox.new',
  'workbench.action.files.saveAs',
  'editor.action.addCommentLine',
  'codesandbox.sandbox.exportZip',
  'codesandbox.preferences',
  'codesandbox.sandbox.fork',
  'codesandbox.help.documentation',
  'codesandbox.help.github',
  'view.preview.flip',
  'codesandbox.search',
  'workbench.action.splitEditorLeft',
  'codesandbox.dashboard',
  'workbench.action.toggleCenteredLayout',
  'workbench.action.toggleMenuBar',
  'codesandbox.explore',
  'editor.action.toggleTabFocusMode',
  'workbench.action.splitEditorUp',
  'workbench.action.toggleSidebarPosition',
  'workbench.action.toggleActivityBarVisibility',
  'workbench.action.toggleStatusbarVisibility',
  'codesandbox.dependencies.add',
  'codesandbox.help.open-issue',
  'codesandbox.action.search',
  'workbench.action.editorLayoutThreeColumns',
  'breadcrumbs.toggleToOn',
  'workbench.action.openSettings2',
  'workbench.action.globalSettings',
  'workbench.action.editorLayoutTwoRows',
  'workbench.action.editorLayoutTwoByTwoGrid',
  'editor.action.showContextMenu',
  'toggleVim',
  'codesandbox.help.spectrum',
  'codesandbox.help.feedback',
  'workbench.action.webview.openDeveloperTools',
  'workbench.action.editorLayoutThreeRows',
  'codesandbox.help.twitter',
  'workbench.action.editorLayoutTwo',
  'codesandbox.preview.external',
  'notifications.showList',
  'workbench.action.editor.changeEncoding',
  'editor.action.indentationToTabs',
  'workbench.action.maximizeEditor',
  'editor.action.indentationToSpaces',
  'revealFilesInOS',
  'keybindings.editor.searchKeyBindings',
  'notifications.hideList',
  'workbench.action.terminal.focus',
  'workbench.action.console.focus',
  'workbench.action.openRecent',
  'code-runner.run',
];

export const DNT =
  typeof window !== 'undefined' &&
  Boolean(
    global.doNotTrack === '1' ||
      global.navigator.doNotTrack === '1' ||
      global.navigator.msDoNotTrack === '1'
  );

let sentryInitialized = false;

function getSentry() {
  return import(/* webpackChunkName: 'sentry' */ '@sentry/browser');
}
export async function initializeSentry(dsn: string) {
  if (!DNT) {
    sentryInitialized = true;
    const Sentry = await getSentry();

    return Sentry.init({
      dsn,
      release: VERSION,
      ignoreErrors: [
        'Custom Object', // Called for errors coming from sandbox (https://sentry.io/organizations/codesandbox/issues/965255074/?project=155188&query=is%3Aunresolved&statsPeriod=14d)
        'TypeScript Server Error', // Called from the TSC server
        /^Canceled$/, // Used by VSCode to stop currently running actions
      ],
      /**
       * Don't send messages from the sandbox, so don't send from eg.
       * new.codesandbox.io or new.csb.app
       */
      blacklistUrls: [
        'codesandbox.editor.main.js',
        /.*\.codesandbox\.io/,
        /.*\.csb\.app/,
      ],
      beforeSend: event => {
        if (
          event.stacktrace &&
          event.stacktrace.frames &&
          event.stacktrace.frames[0] &&
          event.stacktrace.frames[0].filename.endsWith(
            'codesandbox.editor.main.js'
          )
        ) {
          // This is the spammy event that doesn't do anything: https://sentry.io/organizations/codesandbox/issues/1054971728/?project=155188&query=is%3Aunresolved
          // Don't do anything with it right now, I can't seem to reproduce it for some reason.
          // We need to add sourcemaps
          return undefined;
        }

        return event;
      },
    });
  }

  return Promise.resolve();
}

export async function logError(err: Error) {
  if (sentryInitialized) {
    const Sentry = await getSentry();
    Sentry.captureException(err);
  }

  if (window.console && console.error) console.error(err);
}

export async function identify(key: string, value: string) {
  try {
    if (!DNT) {
      if (typeof global.amplitude !== 'undefined') {
        const identity = new global.amplitude.Identify();
        identity.set(key, value);
        global.amplitude.identify(identity);
        debug('[Amplitude] Identifying', key, value);
      }

      if (sentryInitialized) {
        const Sentry = await getSentry();

        Sentry.configureScope(scope => {
          scope.setExtra(key, value);
        });
      }
    }
  } catch (e) {
    /* */
  }
}

if (process.env.NODE_ENV === 'production') {
  setTimeout(() => {
    identify('[Amplitude] Version', VERSION);
  }, 5000);
}

export async function setUserId(userId: string) {
  try {
    if (!DNT) {
      if (typeof global.amplitude !== 'undefined') {
        const hashedId = hash(userId);
        debug('[Amplitude] Setting User ID', hashedId);
        identify('userId', hashedId);

        global.amplitude.getInstance().setUserId(hashedId);
      }

      if (sentryInitialized) {
        const Sentry = await getSentry();
        Sentry.configureScope(scope => {
          scope.setUser({ id: userId });
        });
      }
    }
  } catch (e) {
    /* */
  }
}

export async function resetUserId() {
  try {
    if (!DNT) {
      if (typeof global.amplitude !== 'undefined') {
        debug('[Amplitude] Resetting User ID');
        identify('userId', null);

        if (global.amplitude.getInstance().options.userId) {
          global.amplitude.getInstance().setUserId(null);
          global.amplitude.getInstance().regenerateDeviceId();
        }
      }

      if (sentryInitialized) {
        const Sentry = await getSentry();
        Sentry.configureScope(scope => {
          scope.setUser({ id: undefined });
        });
      }
    }
  } catch (e) {
    /* */
  }
}

const isAllowedEvent = (eventName, secondArg) => {
  try {
    if (eventName === 'VSCode - workbenchActionExecuted') {
      return WHITELISTED_VSCODE_EVENTS.includes(secondArg.id);
    }
    return true;
  } catch (e) {
    return true;
  }
};

// After 30min no event we mark a session
const NEW_SESSION_TIME = 1000 * 60 * 30;

const getLastTimeEventSent = () => {
  const lastTime = localStorage.getItem('csb-last-event-sent');

  if (lastTime === null) {
    return 0;
  }

  return +lastTime;
};

const markLastTimeEventSent = () => {
  localStorage.setItem('csb-last-event-sent', Date.now().toString());
};

export default function track(eventName, secondArg: Object = {}) {
  try {
    if (!DNT && isAllowedEvent(eventName, secondArg)) {
      const data = {
        ...secondArg,
        version: VERSION,
        path: location.pathname + location.search,
      };
      try {
        if (global.ga) {
          global.ga('send', data);
        }
      } catch (e) {
        /* */
      }
      try {
        if (typeof global.amplitude !== 'undefined') {
          const currentTime = Date.now();
          if (currentTime - getLastTimeEventSent() > NEW_SESSION_TIME) {
            // We send a separate New Session event if people have been inactive for a while
            global.amplitude.logEvent('New Session');
          }
          markLastTimeEventSent();

          debug('[Amplitude] Tracking', eventName, data);
          global.amplitude.logEvent(eventName, data);
        }
      } catch (e) {
        /* */
      }
    }
  } catch (e) {
    /* empty */
  }
}
