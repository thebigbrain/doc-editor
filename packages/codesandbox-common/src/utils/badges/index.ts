/* @flow */
import Patron4 from './svg/patron-4.svg';
import Patron3 from './svg/patron-3.svg';
import Patron2 from './svg/patron-2.svg';
import Patron1 from './svg/patron-1.svg';

export default function getBadge(badgeId: string): string {
  if (badgeId === 'patron_4') return Patron4;
  if (badgeId === 'patron_3') return Patron3;
  if (badgeId === 'patron_2') return Patron2;
  if (badgeId === 'patron_1') return Patron1;

  return '';
}
