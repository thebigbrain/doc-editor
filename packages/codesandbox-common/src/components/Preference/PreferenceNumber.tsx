import React from 'react';
import { StyledInput } from './elements';

export type Props = {
  setValue: (value: number) => void;
  value: number;
  step?: number;
  style?: React.CSSProperties;
};

export default class PreferenceInput extends React.PureComponent<Props> {
  handleChange = e => {
    const { value } = e.target;

    if (!Number.isNaN(+value)) {
      this.props.setValue(+value);
    }
  };

  render() {
    const { value, style, step } = this.props;

    return (
      <StyledInput
        step={step}
        style={{ width: '3rem', ...style }}
        type="number"
        value={value}
        onChange={this.handleChange}
      />
    );
  }
}
