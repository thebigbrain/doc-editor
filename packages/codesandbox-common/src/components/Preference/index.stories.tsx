import React from 'react';
import { storiesOf } from '@storybook/react';
import { noop } from '../../test/mocks';
import Preference from '.';
import { KEYBINDINGS } from '../../utils/keybindings';

const stories = storiesOf('components/Preference', module);
const keyBindingKeys = Object.keys(KEYBINDINGS);

stories
  .add('Boolean Preference', () => (
    <Preference
      setValue={noop}
      value={false}
      title="Vim Mode?"
      type="boolean"
    />
  ))
  .add('String Preference', () => (
    <Preference
      setValue={noop}
      title="Whats your name?"
      type="string"
      value="Test"
    />
  ))
  .add('Keybinding Preference', () =>
    keyBindingKeys.map((id, i) => (
      <Preference
        setValue={noop}
        key={id}
        title={KEYBINDINGS[id].title}
        value={KEYBINDINGS[id].bindings}
        type="keybinding"
      />
    ))
  )
  .add('Dropdown Preference', () => (
    <Preference
      title="Select your editor"
      setValue={noop}
      type="dropdown"
      value="one"
      options={['one', 'two']}
    />
  ));
