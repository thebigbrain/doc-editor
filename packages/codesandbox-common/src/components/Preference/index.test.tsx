import 'jest-styled-components';
import React from 'react';
import { noop } from '../../test/mocks';
import mountWithTheme from '../../test/themeMount';
import Preference from '.';
import { KEYBINDINGS } from '../../utils/keybindings';

const keyBindingKeys = Object.keys(KEYBINDINGS);

describe('<Preference /> rendering', () => {
  it('boolean', () => {
    const wrapper = mountWithTheme(
      <Preference
        setValue={noop}
        value={false}
        title="Vim Mode?"
        type="boolean"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('string', () => {
    const wrapper = mountWithTheme(
      <Preference
        setValue={noop}
        title="Whats your name?"
        type="string"
        value="Test"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('dropdown', () => {
    const wrapper = mountWithTheme(
      <Preference
        title="Select your editor"
        setValue={noop}
        type="dropdown"
        value="one"
        options={['one', 'two']}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  keyBindingKeys.map((id, i) =>
    it('string', () => {
      const wrapper = mountWithTheme(
        <Preference
          setValue={noop}
          key={id}
          title={KEYBINDINGS[id].title}
          value={KEYBINDINGS[id].bindings}
          type="keybinding"
        />
      );
      expect(wrapper).toMatchSnapshot();
    })
  );
});
