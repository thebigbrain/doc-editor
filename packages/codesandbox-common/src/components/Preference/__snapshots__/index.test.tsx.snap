// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preference /> rendering boolean 1`] = `ReactWrapper {}`;

exports[`<Preference /> rendering dropdown 1`] = `ReactWrapper {}`;

exports[`<Preference /> rendering string 1`] = `ReactWrapper {}`;

exports[`<Preference /> rendering string 2`] = `ReactWrapper {}`;

exports[`<Preference /> rendering string 3`] = `ReactWrapper {}`;

exports[`<Preference /> rendering string 4`] = `ReactWrapper {}`;

exports[`<Preference /> rendering string 5`] = `ReactWrapper {}`;

exports[`<Preference /> rendering string 6`] = `ReactWrapper {}`;

exports[`<Preference /> rendering string 7`] = `ReactWrapper {}`;

exports[`<Preference /> rendering string 8`] = `ReactWrapper {}`;

exports[`<Preference /> rendering string 9`] = `ReactWrapper {}`;

exports[`<Preference /> rendering string 10`] = `ReactWrapper {}`;

exports[`<Preference /> rendering string 11`] = `ReactWrapper {}`;
